### PR TITLE
improvement: TxSummary

### PR DESCRIPTION
**Pull request by NunoAlexandre**
### Changes
- turn the `FeePreview` component into `TxSummary` 
- redesign it a bit
- include the token price in dollars
- include a little warning when the tx amount is a bit too low compared to the fee (force-shown at the screenshot below)

Screenshot 2024-11-14 at 12 59 41

### CODE REVIEW

File: app/src/hooks/useTokenPrice.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { useCallback, useEffect, useState } from 'react'
+
```
Comment (noahjoeris) [inline app/src/hooks/useTokenPrice.tsx:2]:
```suggestion
```

File: app/src/hooks/useTokenPrice.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      return
+    }
+
+    try {
+      const price = (await getTokenPrice(getCoingekoId(token)))?.usd ?? 0
+      setPrice(price)
+    } catch (e) {
+      // Do not throw an error here
+      console.error('useTokenPrice: Failed to fetch with error:', e)
+    }
```
Comment (noahjoeris) [inline app/src/hooks/useTokenPrice.tsx:21]: Adding Sentry log?

**Review by noahjoeris — APPROVED** [review]
Wow really great changes. I love it ❤️

File: app/src/models/transfer.ts [hunk 3]
```diff
@@ -76,12 +76,13 @@ export type CompletedTransfer = {
   errors?: string[]
 }
 export type TransfersByDate = Record<string, CompletedTransfer[]>
-export interface Fees {
+
+export interface AmountInfo {
   /* The amount in the `token` currency */
-  amount: string
-  /* the token of the fees */
+  amount: string | bigint | number
```
Comment (Victor-Salomon) [inline app/src/models/transfer.ts:82]: Did you try to perform a transfer with this new interface? 
(I remember seeing some parsing errors with bigint in the local storage to store the user's transfers).

**Review by Victor-Salomon — APPROVED** [review]
Left a comment but the code looks mint!

File: app/src/models/transfer.ts (hunk 3 above)
Comment (NunoAlexandre) [reply app/src/models/transfer.ts:82]: I think I did but let me try again 👍  otherwise we need to find another way of course. Thanks for raising this issue!

File: app/src/hooks/useTokenPrice.tsx (hunk 2 above)
Comment (NunoAlexandre) [reply app/src/hooks/useTokenPrice.tsx:21]: Good catch, thank you!

File: app/src/models/transfer.ts (hunk 3 above)
Comment (NunoAlexandre) [reply app/src/models/transfer.ts:82]: @Victor-Salomon just tried this and it worked just fine 👌  I refreshed the browser to request a fresh read from the storage and there were no issues. Can you checkout this branch and try one as well? Maybe you know better how to break the system :p

Comment (Victor-Salomon) [reply app/src/models/transfer.ts:82]: Sure! I'll try it locally and confirm here it's working well :)

Comment (NunoAlexandre) [reply app/src/models/transfer.ts:82]: Screenshot 2024-11-18 at 12 51 17

**Comment by Victor-Salomon** [comment]
@NunoAlexandre tested and validated! Everything works like a charm 😈
